openhands/integrations/provider.py

```diff
@@ -449,7 +449,10 @@ def get_provider_env_key(cls, provider: ProviderType) -> str:
         return f'{provider.value}_token'.lower()
 
     async def verify_repo_provider(
-        self, repository: str, specified_provider: ProviderType | None = None
+        self,
+        repository: str,
+        specified_provider: ProviderType | None = None,
+        is_optional: bool = False,
```

Comment:
NIT: maybe rename `is_optional` so its more obvious its only related to logging level